eureka_ml_insights/data_utils/transform.py

```diff
@@ -302,3 +304,45 @@ def transform(self, df: pd.DataFrame, encoding="cl100k_base") -> pd.DataFrame:
             token_count_column = f"{column}_token_count"
             df[token_count_column] = token_count
         return df
+
+@dataclass
+class MajorityVoteTransform:
+    """Applies the majority vote transformation to the specified model output column per ID."""
+    
+    model_output_col: str = "model_output"  # Default column name for model outputs
+    id_col: str = "ID"  # Default column name for IDs
+    
+    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Transforms the dataframe by calculating the majority vote of 'model_output' per 'ID'.
+        If the 'model_output' is NaN, it will be droped in the majority vote.
+
+        Args:
+            df (pd.DataFrame): Input dataframe containing 'ID' and 'model_output'.
+        
+        Returns:
+            pd.DataFrame: Transformed dataframe with majority vote for each 'ID'.
+        """
+        
+        # Function to get the majority vote
+        def majority_vote(group):
+            # Drop NaN values and get the most common value
+            group_non_nan = group.dropna()
+            if not group_non_nan.empty:
+                return group_non_nan.mode().iloc[0]  # Return the mode, the most frequent value
+            else:
+                return np.nan # pd.NA  # Return NaN if all values are NaN
+        
+        # Custom function to get the first value, regardless of NA
+        def first_in_group(group):
+            return group.iloc[0]  # Return the first element
+        
+        # Perform aggregation
+        other_columns = [col for col in df.columns if col != self.id_col and col != self.model_output_col]
+
+        df_filtered = df.groupby(self.id_col).agg(
+            {self.model_output_col: majority_vote,  # Aggregating 'model_output' using majority vote
```

Comment:
If you could take one of your existing inference results and evaluate it using the latest version of the code, and make sure you get the same results, that would be great. I'm afraid you won't even get the same number of rows as before, and the result will be different.

Reply:
simplify the majority vote